table_to_json.py

```diff
@@ -4,82 +4,97 @@
-'''
+class TableToJSON:
+    '''
+    Converts table from html document to json format
+    '''
 
-#return matrix maked from table for specified tag.
-def create_matrix(tag_type):
-    temporary_table = np.empty(shape=(len(converted_tr), table_width), dtype=object)
-    for idx, single_tr in enumerate(converted_tr):
-        column_shift = 0  # when encouter colspan add number
-        for idxtd, single_td in enumerate(single_tr.find_all(lambda tag: tag.name == 'td' or tag.name == 'th')):
-            if single_td.name == tag_type:
-                while temporary_table[idx, idxtd + column_shift] is not None:
-                    column_shift += 1
+    def __init__(self, file_name):
+        with open(file_name) as fp:
+            soup = BeautifulSoup(fp.read(), "lxml")
+        self.converted_tr = soup.find_all('tr')
+        self.table_width = 0
 
-                colspan_range = int(single_td.get("colspan", "1"))
-                rowspan_range = int(single_td.get("rowspan", "1"))
+        # find width of the table
+        for single_td in self.converted_tr[0].find_all('th'):
+            self.table_width += int(single_td.get("colspan", "1"))
 
-                if colspan_range == 1 and rowspan_range == 1:
-                    temporary_table[idx, idxtd + column_shift] = single_td.text.strip()
-                else:
-                    for j in range(rowspan_range):
-                        for i in range(colspan_range):
-                            while temporary_table[idx + j, idxtd + column_shift + i]:
-                                column_shift += 1
-                            temporary_table[idx + j, idxtd + column_shift + i] = single_td.text.rstrip()
-                    column_shift += int(colspan_range) - 1
-    return temporary_table
+        # add only td to table named temporary_table_td
+        self.temporary_table_td = TableToJSON.create_matrix(self, 'td')
 
+        # add only th to table named temporary_table_th
+        self.temporary_table_th = TableToJSON.create_matrix(self, 'th')
 
-# search for th in each matrix row to the left from td (is_row) or in each column above td
-def search_th(search_range, pos_td, is_row):
-    temp_name = ''
-    for  s in range(search_range):
-     if is_row:
-         second_pos, first_pos = s, pos_td
-     else:
-         second_pos, first_pos = pos_td, s
 
-     if temporary_table_th[first_pos,  second_pos]:
-        temp_name = temp_name + temporary_table_th[first_pos,  second_pos] + '.'
-    return temp_name
+    #return matrix maked from table for specified tag.
+    def create_matrix(self, tag_type):
+        temporary_table = np.empty(shape=(len(self.converted_tr), self.table_width), dtype=object)
+        for idx, single_tr in enumerate(self.converted_tr):
+            column_shift = 0  # when encouter colspan add number
+            for idxtd, single_td in enumerate(single_tr.find_all(lambda tag: tag.name == 'td' or tag.name == 'th')):
```

Comment:
ja jednak twierdzę, że się da to samo zrobić bez lambdy, or-a itp. patrz: https://stackoverflow.com/a/20649408